src/main/java/org/ow2/proactive/catalog/service/BucketGrantService.java

```diff
@@ -407,4 +403,31 @@ public void deleteAllBucketGrants(long bucketId) {
     public long getBucketIdByName(String bucketName) {
         return bucketRepository.findOneByBucketName(bucketName).getId();
     }
+
+    public String getHighestGrantAccessTypeFromBucketGrants(AuthenticatedUser user, String bucketName,
+            String bucketOwner) {
+        String accessType = "";
+        if (user.getName().equals(bucketOwner) || user.getGroups().contains(bucketOwner.substring(6)) ||
+            bucketOwner.equals("GROUP:public-objects")) {
+            accessType = admin.toString();
+        } else {
+            long bucketId = bucketRepository.findOneByBucketName(bucketName).getId();
+            List<BucketGrantMetadata> grants = this.getAllAssignedGrantsForUserAndHisGroups(user.getName(),
+                                                                                            user.getGroups());
+            grants = grants.stream().filter(grant -> grant.getBucketId() == bucketId).collect(Collectors.toList());
+            if (grants.size() > 0) {
```

Comment:
is it possible that grants is Null ? If yes, can you add a check here ?

Reply:
The way I did getAllAssignedGrants..., it will return an empty list in case of null or no grants.